src/runtime/ListWrapper.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Python.Runtime {
+    /// <summary>
+    /// Implements IEnumerable<typeparamref name="T"/> for any python iterable.
+    /// </summary>
+    internal class IterableWrapper<T> : IEnumerable<T> {
+        private IntPtr iterObject;
+
+        public IterableWrapper(IntPtr value) {
+            iterObject = Runtime.PyObject_GetIter(value);
+            if (iterObject == IntPtr.Zero)
+                Exceptions.RaiseTypeError("not an iterator");
+            Runtime.XIncref(iterObject);
+        }
+        ~IterableWrapper() {
+            Runtime.XDecref(iterObject);
```

Comment:
You can't use `XDecref` directly in destructor, see `Runtime.PyObject`'s destructor, you can just refer a `PyObject`, it can let you have no need concern about about the release.